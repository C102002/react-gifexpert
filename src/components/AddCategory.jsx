import { useState } from "react"
import PropTypes from 'prop-types';

//NT: Los accidentes no existen
export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputvalue] = useState('One punch');

  const onInputChange=(event)=>{
    //console.log(event.target.value)
    setInputvalue(event.target.value)
  }

  const onSubmit=(event)=>{
    event.preventDefault()
    const input=inputValue

    //NT: Al usar categories asi, no perdemos la referencia de los datos de antes
    //NT:validacion de string<=1
    if(input.trim().length<=1) return;
    //NT:Mandandolo de forma de evento al padre
    onNewCategory(input);
    //NT:Reiniciandolo
    setInputvalue('');
  }
    return (
        //NT: Manda la info aca cuando le da enter
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text"
            placeholder="Buscar gif"
            value={inputValue}
            onChange={(event)=>onInputChange(event)}
            // Es lo mismo de lo de arriba onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}