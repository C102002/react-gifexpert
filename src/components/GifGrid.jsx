import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const {images,isLoading}=useFetchGifs(category);
    //Nt: como era antes sin el custom hook
    // const [images, setImages] = useState([])

    // const getImages= async()=>{
    //     const newImages=await getGifs(category);
    //     setImages(newImages)
    // }

    // //NT: UseEfect es para hacer efectos secundarios, el [] es el que hace que se haga solamente una vez
    // useEffect(()=>{
    //     getGifs(category)
    //     getImages();
    // },[])

   
    getGifs(category);

  return (
    <>
    <h3>{category}</h3>
    {
        //NT: isLoading si es true se renderiza el h2
        isLoading && (<h2>Cargando...</h2>)
    }
    <div className='card-grid'>
        {
            images.map((image)=>(
                <GifItem 
                key={image.id}
                title={image.title}
                url={image.url}
                //NT: se puede hacer asi {... image} para sacar todas las props
                />
            ))
        }
    </div>
    </>
  )
}

GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}