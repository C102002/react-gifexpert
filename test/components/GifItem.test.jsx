import { render,screen } from "@testing-library/react"
import {GifItem} from "../../src/components/GifItem"



describe('Prueba de GifItem', () => {
    const title='Saitama'
    const url ='https://one-punch.com/saitama.jpg'

    test('1.Prueba de los props (titulo obligatorio y url obligatorio)', () => { 
        const {container}= render (<GifItem title={title} url={url} />)
    })
    test('2. Prueba del snapshot', () => {
        const {container}= render (<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })

     test('3. Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render (<GifItem title={title} url={url}/>)
        //NT: Es lo mismo pero mas ordenado
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src,alt} =screen.getByRole('img');
        console.log(src)
        expect(src).toBe(url);
        expect(alt).toBe(title);
     })

     test('4.El titulo este mostrado como  un texto', () => {
        render (<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
      })
 })