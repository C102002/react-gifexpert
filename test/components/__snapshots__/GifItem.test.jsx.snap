// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifItem 2. Prueba del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
